register-machines/Cargo.toml

```diff
@@ -6,5 +6,4 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-num-bigint = "0.4.3"
-num-traits = "0.2.14"
\ No newline at end of file
+rug = "1.14.0"
```

Comment:
I really like the fact that you've found a different library for arbitrary-ranged integers!